File: lib/app/Models/PromoOfferDiscountParamsDTO.php
<?php
/**
 * PromoOfferDiscountParamsDTO
 */
namespace app\Models;

/**
 * PromoOfferDiscountParamsDTO
 * @description Параметры товара в акции с типом `DIRECT_DISCOUNT` или `BLUE_FLASH`.
 */
class PromoOfferDiscountParamsDTO {

    /** @var int $price Зачеркнутая цена — та, по которой товар продавался до акции.  Указывается в рублях.  Возвращается, только если товар участвует в акции.*/
    public $price = 0;

    /** @var int $promoPrice Цена по акции — та, по которой вы хотите продавать товар.  Указывается в рублях.  Возвращается, только если товар участвует в акции.*/
    public $promoPrice = 0;

    /** @var int $maxPromoPrice Максимально возможная цена для участия в акции.  Указывается в рублях.  Возвращается для всех товаров.*/
    public $maxPromoPrice = 0;

}

File: lib/Models/PromoOfferDiscountParamsDTO.cpp


#include "PromoOfferDiscountParamsDTO.h"

using namespace Tiny;

PromoOfferDiscountParamsDTO::PromoOfferDiscountParamsDTO()
{
	price = long(0);
	promoPrice = long(0);
	maxPromoPrice = long(0);
}

PromoOfferDiscountParamsDTO::PromoOfferDiscountParamsDTO(std::string jsonString)
{
	this->fromJson(jsonString);
}

PromoOfferDiscountParamsDTO::~PromoOfferDiscountParamsDTO()
{

}

void
PromoOfferDiscountParamsDTO::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *priceKey = "price";

    if(object.has_key(priceKey))
    {
        bourne::json value = object[priceKey];



        jsonToValue(&price, value, "long");


    }

    const char *promoPriceKey = "promoPrice";

    if(object.has_key(promoPriceKey))
    {
        bourne::json value = object[promoPriceKey];



        jsonToValue(&promoPrice, value, "long");


    }

    const char *maxPromoPriceKey = "maxPromoPrice";

    if(object.has_key(maxPromoPriceKey))
    {
        bourne::json value = object[maxPromoPriceKey];



        jsonToValue(&maxPromoPrice, value, "long");


    }


}

bourne::json
PromoOfferDiscountParamsDTO::toJson()
{
    bourne::json object = bourne::json::object();





    object["price"] = getPrice();






    object["promoPrice"] = getPromoPrice();






    object["maxPromoPrice"] = getMaxPromoPrice();



    return object;

}

long
PromoOfferDiscountParamsDTO::getPrice()
{
	return price;
}

void
PromoOfferDiscountParamsDTO::setPrice(long  price)
{
	this->price = price;
}

long
PromoOfferDiscountParamsDTO::getPromoPrice()
{
	return promoPrice;
}

void
PromoOfferDiscountParamsDTO::setPromoPrice(long  promoPrice)
{
	this->promoPrice = promoPrice;
}

long
PromoOfferDiscountParamsDTO::getMaxPromoPrice()
{
	return maxPromoPrice;
}

void
PromoOfferDiscountParamsDTO::setMaxPromoPrice(long  maxPromoPrice)
{
	this->maxPromoPrice = maxPromoPrice;
}



